### PR TITLE
Port yuzu-emu/yuzu#1137: "renderer_opengl: Namespace OpenGL code"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See https://github.com/yuzu-emu/yuzu/pull/1137 for more details.

This PR was basically remade as the code of Citra is quite different from yuzu. 
Please point out any places where I set the namespaces wrong, or where this PR can be improved in general.

**Original description:**
Namespaces all OpenGL code under the OpenGL namespace.

Prevents polluting the global namespace and allows clear distinction
between other renderers' code in the future.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4423)

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 1]
```diff
@@ -13,17 +13,15 @@
 #include "common/common_types.h"
 #include "video_core/renderer_opengl/gl_shader_decompiler.h"
 
-namespace Pica {
-namespace Shader {
-namespace Decompiler {
+namespace OpenGL::Pica::Shader::Decompiler {
```
Comment (wwylele) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: I don't know how you feel but this looks too long to me and has unnecessary multiple levels. `OpenGL::ShaderDecompiler` is good enough.

File: src/video_core/renderer_opengl/gl_shader_gen.cpp [hunk 2]
```diff
@@ -118,7 +118,7 @@ PicaFSConfig PicaFSConfig::BuildFromRegs(const Pica::Regs& regs) {
 
     state.alpha_test_func = regs.framebuffer.output_merger.alpha_test.enable
                                 ? regs.framebuffer.output_merger.alpha_test.func.Value()
-                                : Pica::FramebufferRegs::CompareFunc::Always;
+                                : ::Pica::FramebufferRegs::CompareFunc::Always;
```
Comment (wwylele) [inline src/video_core/renderer_opengl/gl_shader_gen.cpp]: And this namespace pollution is caused by your ugly long namespace above...

File: src/video_core/renderer_opengl/gl_shader_manager.h [hunk 3]
```diff
@@ -24,6 +24,8 @@ struct LightSrc {
     GLfloat dist_atten_scale;
 };
 
+namespace OpenGL::GLShader {
```
Comment (wwylele) [inline src/video_core/renderer_opengl/gl_shader_manager.h]: Why are `UniformBindings` and `LightSrc` not included?

File: src/video_core/renderer_opengl/gl_shader_util.cpp [hunk 4]
```diff
@@ -8,7 +8,7 @@
 #include "common/logging/log.h"
 #include "video_core/renderer_opengl/gl_shader_util.h"
 
-namespace GLShader {
+namespace OpenGL::GLShader {
```
Comment (wwylele) [inline src/video_core/renderer_opengl/gl_shader_util.cpp]: The original `GLShader` namespace already implies OpenGL itself. Just replace the name if you prefer `OpenGL`

File: src/video_core/renderer_opengl/renderer_opengl.h [hunk 5]
```diff
@@ -32,6 +30,8 @@ struct ScreenInfo {
     TextureInfo texture;
 };
 
+namespace OpenGL {
```
Comment (wwylele) [inline src/video_core/renderer_opengl/renderer_opengl.h]: Why are `TextureInfo` and `ScreenInfo` not included?

**Comment by FearlessTobi** [comment]
Comments addressed.

**Pull request merged**